<a href="https://colab.research.google.com/github/yuhui-0611/SEED/blob/main/SEED_YB_%EB%8D%B0%EC%9D%B4%ED%84%B0.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving 전체_통합.xlsx to 전체_통합.xlsx


In [ ]:
import pandas as pd
df = pd.read_excel('전체_통합.xlsx', engine='openpyxl')

In [ ]:
df.head()

행정구역    연도_분기   전체점포수  프랜차이즈 점포수  일반 점포수  개업률  폐업률  1년 생존율  3년 생존율  \
0  서울시 전체  2015-Q1  571434      38561  532873  3.1  3.9    75.0    51.1   
1     종로구  2015-Q1   25276       1306   23970  2.5  3.4    80.0    57.0   
2      중구  2015-Q1   43310       1642   41668  2.7  3.7    81.2    59.5   
3     용산구  2015-Q1   18912        969   17943  2.9  3.6    78.8    56.3   
4     성동구  2015-Q1   15655       1030   14625  3.2  3.7    77.8    55.3   

   5년 생존율     log인구   인구(raw)   환산임대료  10년영업기간  30년영업기간   실업률  소비자심리지수(평균)  
0    36.8  7.015740  10369067   96760      3.6      5.3  4.07   102.966667  
1    42.2  5.218115    165240  151285      3.8      6.1  4.07   102.966667  
2    44.1  5.133740    136063  191889      4.0      6.0  4.07   102.966667  
3    41.7  5.397945    250003   95330      3.8      5.5  4.07   102.966667  
4    40.2  5.482555    303777   76623      3.7      5.6  4.07   102.966667

---
## [1] 변수 선택

In [ ]:
df[['실업률', '소비자심리지수(평균)']].corr()

실업률  소비자심리지수(평균)
실업률          1.000000    -0.011537
소비자심리지수(평균) -0.011537     1.000000

-> 실업률과 소비자심리지수(평균)은 거의 상관없는 데이터들이라 함께 사용 가능 할듯

In [ ]:
data = df.drop(['행정구역','연도_분기'], axis=1)

In [ ]:
data.corr()

전체점포수  프랜차이즈 점포수    일반 점포수       개업률       폐업률    1년 생존율  \
전체점포수        1.000000   0.995666  0.999968 -0.025883 -0.046070  0.039073   
프랜차이즈 점포수    0.995666   1.000000  0.994888 -0.018881 -0.053224  0.043419   
일반 점포수       0.999968   0.994888  1.000000 -0.026476 -0.045434  0.038682   
개업률         -0.025883  -0.018881 -0.026476  1.000000  0.242583 -0.317601   
폐업률         -0.046070  -0.053224 -0.045434  0.242583  1.000000 -0.636423   
1년 생존율       0.039073   0.043419  0.038682 -0.317601 -0.636423  1.000000   
3년 생존율       0.044198   0.052985  0.043423 -0.263233 -0.705635  0.947629   
5년 생존율       0.047506   0.058231  0.046562 -0.251016 -0.753817  0.905107   
log인구        0.865187   0.872456  0.864209  0.105782  0.134227 -0.204827   
인구(raw)      0.991687   0.984548  0.991900  0.003672  0.002795 -0.022576   
환산임대료        0.100286   0.083430  0.101699 -0.280053 -0.483179  0.656389   
10년영업기간      0.012787  -0.005984  0.014400 -0.067928 -0.317297  0.508200   
30년영업기간      0.013575  -0.014770  0.016013 -0.367416 -0.367774  0.544983   
실업률         -0.014622  -0.025591 -0.013670  0.158163  0.270251 -0.392348   
소비자심리지수(평균) -0.006474  -0.019146 -0.005379 -0.054672  0.287073 -0.198377   

               3년 생존율    5년 생존율     log인구   인구(raw)     환산임대료   10년영업기간  \
전체점포수        0.044198  0.047506  0.865187  0.991687  0.100286  0.012787   
프랜차이즈 점포수    0.052985  0.058231  0.872456  0.984548  0.083430 -0.005984   
일반 점포수       0.043423  0.046562  0.864209  0.991900  0.101699  0.014400   
개업률         -0.263233 -0.251016  0.105782  0.003672 -0.280053 -0.067928   
폐업률         -0.705635 -0.753817  0.134227  0.002795 -0.483179 -0.317297   
1년 생존율       0.947629  0.905107 -0.204827 -0.022576  0.656389  0.508200   
3년 생존율       1.000000  0.983123 -0.177845 -0.016651  0.631376  0.543633   
5년 생존율       0.983123  1.000000 -0.164887 -0.013286  0.620393  0.498591   
log인구       -0.177845 -0.164887  1.000000  0.895605 -0.232961 -0.237707   
인구(raw)     -0.016651 -0.013286  0.895605  1.000000  0.015386 -0.030935   
환산임대료        0.631376  0.620393 -0.232961  0.015386  1.000000  0.420698   
10년영업기간      0.543633  0.498591 -0.237707 -0.030935  0.420698  1.000000   
30년영업기간      0.467598  0.429298 -0.345409 -0.041609  0.626057  0.589849   
실업률         -0.346602 -0.309757  0.020384  0.005822 -0.235375  0.093548   
소비자심리지수(평균) -0.297130 -0.315848  0.008630  0.002632 -0.115968 -0.181999   

              30년영업기간       실업률  소비자심리지수(평균)  
전체점포수        0.013575 -0.014622    -0.006474  
프랜차이즈 점포수   -0.014770 -0.025591    -0.019146  
일반 점포수       0.016013 -0.013670    -0.005379  
개업률         -0.367416  0.158163    -0.054672  
폐업률         -0.367774  0.270251     0.287073  
1년 생존율       0.544983 -0.392348    -0.198377  
3년 생존율       0.467598 -0.346602    -0.297130  
5년 생존율       0.429298 -0.309757    -0.315848  
log인구       -0.345409  0.020384     0.008630  
인구(raw)     -0.041609  0.005822     0.002632  
환산임대료        0.626057 -0.235375    -0.115968  
10년영업기간      0.589849  0.093548    -0.181999  
30년영업기간      1.000000 -0.104523     0.037757  
실업률         -0.104523  1.000000    -0.011537  
소비자심리지수(평균)  0.037757 -0.011537     1.000000

✅ 매우 강한 양의 상관관계
전체점포수 ↔ 일반 점포수: 0.999968

전체점포수 ↔ 프랜차이즈 점포수: 0.995666

3년 생존율 ↔ 5년 생존율: 0.983123

1년 생존율 ↔ 3년 생존율: 0.947629

log인구 ↔ 인구(raw): 0.895605


✅ 강한 음의 상관관계
폐업률 ↔ 5년 생존율: -0.753817

폐업률 ↔ 3년 생존율: -0.705635

→ 폐업률이 높을수록 생존율은 낮다 (당연한 해석).

✅ 중간 정도의 상관관계
환산임대료 ↔ 5년 생존율: 0.620393

환산임대료 ↔ 30년영업기간: 0.626057

→ 임대료가 높은 지역일수록 장기 운영 점포가 많을 가능성.

10년영업기간 ↔ 30년영업기간: 0.589849

→ 장기운영 점포는 중기운영 점포와도 상관관계를 보임.

---
- 해결방안
1. 중복되는 변수 중 하나 제거
예: 전체점포수, 일반 점포수, 프랜차이즈 점포수는 서로 상관이 매우 높음.
→ 하나만 선택 (예: 전체점포수).3년 생존율, 5년 생존율, 1년 생존율도 높은 상관
-> 3년 삭제 하는게 좋을 듯, 전체점포수, 일반 점포수, 프랜차이즈 점포수는 비율 사용

2. 지표를 결합해 하나의 변수로 축소
예: 10년, 30년 영업기간 → 장기점포비율 같은 새로운 변수로 축소 가능

3. 표준화 및 주성분 분석 (PCA)
정량적으로 줄여야 할 때 PCA로 차원 축소해 공선성 제거 가능 (단 해석은 어려워짐)

4. 다중공선성 진단 (VIF: Variance Inflation Factor)
statsmodels나 linearmodels에서 VIF를 계산해 기준값(VIF > 10 등) 초과 변수 제거 고려

In [ ]:
df['프랜차이즈비율'] = df['프랜차이즈 점포수']/df['전체점포수']
df = df.drop(['인구(raw)','전체점포수','프랜차이즈 점포수','일반 점포수'], axis=1)

In [ ]:
df.head()

행정구역    연도_분기  개업률  폐업률  1년 생존율  3년 생존율  5년 생존율     log인구   환산임대료  \
0  서울시 전체  2015-Q1  3.1  3.9    75.0    51.1    36.8  7.015740   96760   
1     종로구  2015-Q1  2.5  3.4    80.0    57.0    42.2  5.218115  151285   
2      중구  2015-Q1  2.7  3.7    81.2    59.5    44.1  5.133740  191889   
3     용산구  2015-Q1  2.9  3.6    78.8    56.3    41.7  5.397945   95330   
4     성동구  2015-Q1  3.2  3.7    77.8    55.3    40.2  5.482555   76623   

   10년영업기간  30년영업기간   실업률  소비자심리지수(평균)   프랜차이즈비율  
0      3.6      5.3  4.07   102.966667  0.067481  
1      3.8      6.1  4.07   102.966667  0.051670  
2      4.0      6.0  4.07   102.966667  0.037913  
3      3.8      5.5  4.07   102.966667  0.051237  
4      3.7      5.6  4.07   102.966667  0.065794

---

## [2] 정규화 및 표준화
정규화: 범위가 큰 값 (인구(raw), 환산임대료, 실업률, 소비자심리지수(평균))

표준화: 비율 값이나 퍼센트 값 (개업률, 폐업률, 1년 생존율, 3년 생존율, 5년 생존율, 10년영업기간, 30년영업기간)

In [ ]:
df2 = df.copy()
from sklearn.preprocessing import StandardScaler, MinMaxScaler
normalize_cols = ['인구(raw)', '환산임대료', '실업률', '소비자심리지수(평균)']
standardize_cols = ['개업률', '폐업률', '1년 생존율', '3년 생존율', '5년 생존율', '10년영업기간', '30년영업기간']

minmax_scaler = MinMaxScaler()
standard_scaler = StandardScaler()

df2[normalize_cols] = minmax_scaler.fit_transform(df2[normalize_cols])
df2[standardize_cols] = standard_scaler.fit_transform(df2[standardize_cols])
df2.head()

행정구역    연도_분기   전체점포수  프랜차이즈 점포수  일반 점포수       개업률       폐업률    1년 생존율  \
0  서울시 전체  2015-Q1  571434      38561  532873 -0.205034  1.502138 -0.599783   
1     종로구  2015-Q1   25276       1306   23970 -0.858796  0.672844  0.807332   
2      중구  2015-Q1   43310       1642   41668 -0.640875  1.170420  1.145040   
3     용산구  2015-Q1   18912        969   17943 -0.422955  1.004562  0.469625   
4     성동구  2015-Q1   15655       1030   14625 -0.096074  1.170420  0.188202   

     3년 생존율    5년 생존율     log인구   인구(raw)     환산임대료   10년영업기간   30년영업기간  \
0 -0.669821 -0.866426  7.015740  1.000000  0.150950  1.494381 -0.031013   
1  0.351861 -0.001094  5.218115  0.003367  0.387711  2.523282  2.403180   
2  0.784777  0.303375  5.133740  0.000518  0.564024  3.552184  2.098906   
3  0.230645 -0.081217  5.397945  0.011646  0.144740  2.523282  0.577536   
4  0.057478 -0.321587  5.482555  0.016899  0.063510  2.008831  0.881810   

        실업률  소비자심리지수(평균)  
0  0.678571     0.734314  
1  0.678571     0.734314  
2  0.678571     0.734314  
3  0.678571     0.734314  
4  0.678571     0.734314

---
## [3] 논문 목차
1. 기초통계
- 상관행렬: 주요 변수 간 선형관계의 강도를 파악
- VIF(분산팽창인자): 변수별 VIF를 계산해 다중공선성이 심각한지(일반적으로 VIF > 10) 확인


2. 모형 선택 검정
  
  (1) F-검정 (poolability test)
  - 목적: Pooled OLS(=공통계수) vs 개체별(자치구) 고정효과 모형
  - Stata의 xttest0(pF test)나 R의 plm::pFtest() 사용

  (2)Hausman 검정
  - 목적: 고정효과(FE) vs 확률효과(RE) 중 일관성 비교
  - H₀: RE 일관성 유지 → p-value < 0.05면 FE 선택

  (3) 1계 자기상관 검정
  - Wooldridge test: 패널 FE/RE 모형에서 1차 자기상관 존재 여부
  - H₀: 자기상관 없음 → plm::pwartest() 또는 Stata xtserial
  - 해결: 자기상관이 확인되면 GLS (xtregar) 또는 차분 모형(FD estimator) 고려

  (4) 이분산성 검정
  - Modified Wald test for groupwise heteroskedasticity in FE models
  - Stata xttest3, R plm::bptest(..., group = "group") 등


3. 고정효과 회귀 모형 추정
- Demeaning: 각 변수에서 개체별 평균(자치구별 평균)을 차감한 뒤 OLS 수행


4. 결과 해석
- βk (지역변수): 각 자치구 특성(예: 임대료, 프랜차이즈비율)이 폐업률에 미치는 한 단위 변화 효과
- βs (거시변수): 실업률·소비자심리지수 같은 경기변수가 폐업률에 미치는 동시(혹은 지연)효과
- 통계적 유의성: p-value,
- 모형설명력 비교: within R² 변화, AIC/BIC 값 비교


5. Robustness Checks (강건성 검토)
- 논문에서는 통상 다음 추가 분석으로 결과의 안정성을 확인
- 대체 사양: Random Effects나 Pooled OLS, First-Difference 추정치 비교
- 지연효과 포함: 거시·상권변수의 1분기(lag = 1) 지연항 추가
- 서브샘플: 강남권 vs 도심 외곽, 고임대료 vs 저임대료 등으로 분석 반복


- 참고 문헌:
https://www-dbpia-co-kr-ssl.access.ewha.ac.kr/journal/articleDetail?nodeId=NODE10583689
https://www-dbpia-co-kr-ssl.access.ewha.ac.kr/journal/articleDetail?nodeId=NODE10558908
-> 공통적으로 하우스만 검정, 강건성 검토, Pooled OLS,RE,FE 중 어떤 모델을 사용해야 할지 검정 등등